app/uk/gov/hmrc/nisp/views/account_mqp.scala.html

```diff
@@ -56,72 +57,71 @@ <h2 class="heading-medium">@Messages("nisp.main.description.mqp") @Dates.formatD
         </p>
     </div>
 
-    @if(canGetPension) {
-        <h2 class="heading-medium">@Messages("nisp.mqp.possible")</h2>
-        <p>
-            @if(spSummary.yearsToContributeUntilPensionAge > 0) {
-                @Messages("nisp.mqp.youNeedToAdd",
-                Time.years(numberOfYearsMissing),
-                Time.years(spSummary.yearsToContributeUntilPensionAge),
-                Dates.formatDate(spSummary.statePensionAge.date.localDate))
-            } else { 
-                @Messages("nisp.mqp.youNeedToAddZeroLeft",
-                Time.years(numberOfYearsMissing),
-                Dates.formatDate(spSummary.statePensionAge.date.localDate))
-            }
-        </p>
+    @spSummary.mqp match {
+        case Some(CanGetWithGaps) => {
+            <h2 class="heading-medium">@Messages("nisp.mqp.possible")</h2>
+            <p>
+                @if(spSummary.yearsToContributeUntilPensionAge > 0) {
+                    @Messages("nisp.mqp.youNeedToAdd",
+                    Time.years(numberOfYearsMissing),
+                    Time.years(spSummary.yearsToContributeUntilPensionAge),
+                    Dates.formatDate(spSummary.statePensionAge.date.localDate))
+                } else {
+                    @Messages("nisp.mqp.youNeedToAddZeroLeft",
+                    Time.years(numberOfYearsMissing),
+                    Dates.formatDate(spSummary.statePensionAge.date.localDate))
+                }
+            </p>
+
+            <p>@Messages("nisp.mqp.years.dontcount",Time.years(spSummary.numberOfGapsPayable))</p>
+
+            <p> @Messages("nisp.mqp.afterspa", Dates.formatDate(spSummary.statePensionAge.date.localDate)) </p>
+            <a href='@routes.NIRecordController.showGaps.url' role="button">@Messages("nisp.main.context.fillGaps.viewGaps")</a>
+            <div class="panel-indent">
+                <p>
+                    @if(spSummary.isAbroad) {
+                        @Html(Messages("nisp.main.overseas"))
+                    } else {
+                        @Html(Messages("nisp.mqp.overseas"))
+                    }
+                </p>
+            </div>
+        }
+
+        case _ => {
```

Comment:
Can we change this case to case `Some(CantGet)` and then add a case underneath for `_`. This can just Log an error and display nothing because it shouldn't happen. I think it would be better than telling people they can't get any pension when they can!